### PR TITLE
Minimal over() implementation for aggregate functions

### PR DESCRIPTION
**Pull request by MacDue**
This PR adds minimal support for the `OVER()` clause (i.e support for `OVER()` with no arguments).
There are some more advanced features yet to be implemented described here: https://www.sqlservercentral.com/articles/understanding-the-over-clause

There implemented syntax is as follows:
```cpp
 select(active_quest_tbl.hp, count(active_quest_tbl.id).over().as(total_quests))
.from(active_quest_tbl)
.limit(3u)
.offset(1u);
```
Which is serialized to:
 ```sql
SELECT hp, COUNT(id) OVER() as total_quests 
from active_quests 
LIMIT 3 OFFSET 1;
```

Note:
```cpp
 select(active_quest_tbl.hp, count(active_quest_tbl.id).over())
```
Also works, in this case the name of the selection will be the aggregate function name (count in this example). 

This addresses #315

### CODE REVIEW

**Comment by MacDue** [comment]
I'll look into the error (seems like whatever it is is not an issue in gcc 9)

**Comment by rbock** [comment]
Hi @MacDue 

Very nice work! Would be great if you could add a test or two in `test_serializer`.

Best,

Roland

**Comment by MacDue** [comment]
@rbock https://travis-ci.org/rbock/sqlpp11/jobs/650559734?utm_medium=notification&utm_source=github_status

Are auto aliases not applied at this point in serialization?

**Comment by MacDue** [comment]
Added a little test for serializing .over() both with & without an alias.

File: test_serializer/Over.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "Sample.h"
+#include "compare.h"
+#include <sqlpp11/sqlpp11.h>
+
+SQLPP_ALIAS_PROVIDER(dueutil)
```
Comment (rbock) [inline test_serializer/Over.cpp]: Just curious. Is this a typo? Or am I missing some reference?

**Comment by rbock** [comment]
With the possible exception of the one comment, this looks very good to me.

Let me know if you think it is ready to be merged, too.

File: test_serializer/Over.cpp (hunk 1 above)
Comment (MacDue) [reply test_serializer/Over.cpp]: I just randomly put the name of the project I added this for the test. Don't mind changing it to something else if you like.

**Comment by MacDue** [comment]
I think it's pretty much ready. Maybe some thought should be spared for the other features of `OVER()` at a later date, but it'd require quite a bit of extra work (and would probably just be adding stuff for completeness).

File: test_serializer/Over.cpp (hunk 1 above)
Comment (rbock) [reply test_serializer/Over.cpp]: Thanks for the explanation :-)

**Comment by rbock** [comment]
Done, thanks again :-)

**Pull request merged**